File: model/src/main/java/ar/edu/itba/paw/models/books/BookSearchOrderBy.java
package ar.edu.itba.paw.models.books;

public enum BookSearchOrderBy {
    PRICE_ASC("COALESCE(d.price, b.price) asc", "COALESCE(d.price, b.price) asc", "price.asc"),
    PRICE_DESC("COALESCE(d.price, b.price) desc", "COALESCE(d.price, b.price) desc", "price.desc"),
    PAGE_COUNT_ASC("page_count asc",  "page_count asc","pageCount.asc"),
    PAGE_COUNT_DESC("page_count desc",  "page_count desc","pageCount.desc"),
    PUBLICATION_DATE_ASC("b.book_id desc", "b.bookId asc" ,"publicationDate.asc"),
    PUBLICATION_DATE_DESC("b.book_id desc", "b.bookId desc" ,"publicationDate.desc"),
    BEST_SELLERS("COUNT(o.book_id) desc", "COUNT(o.book_id) desc" ,null),
    NEW_DEALS("d.start_date desc", "d.start_date desc" ,null);

    private final String columnName;
    private final String modelName;
    private final String messageCode;

    BookSearchOrderBy(String columnName, String modelName, String messageCode){
        this.columnName = columnName;
        this.modelName = modelName;
        this.messageCode = messageCode;
    }

    public String getColumnName(){
        return columnName;
    }

    public String getModelName(){
        return modelName;
    }

    public String getMessageCode(){
        return messageCode;
    }
}

File: frontend/src/app/shared/model/book/bookSearchOrderBy.ts
export enum BookSearchOrderBy {
	PRICE_ASC = "PRICE_ASC",
	PRICE_DESC = "PRICE_DESC",
	PAGE_COUNT_ASC = "PAGE_COUNT_ASC",
	PAGE_COUNT_DESC = "PAGE_COUNT_DESC",
	PUBLICATION_DATE_ASC = "PUBLICATION_DATE_ASC",
	PUBLICATION_DATE_DESC = "PUBLICATION_DATE_DESC",
	BEST_SELLERS = "BEST_SELLERS",
	NEW_DEALS = "NEW_DEALS"
}

export const BookSearchOrderByOptions = [
	{ label: "Publication date: Most recent", value: BookSearchOrderBy.PUBLICATION_DATE_DESC },
	{ label: "Publication date: Oldest", value: BookSearchOrderBy.PUBLICATION_DATE_ASC },
	{ label: "Page count: Descending", value: BookSearchOrderBy.PAGE_COUNT_DESC },
	{ label: "Page count: Ascending", value: BookSearchOrderBy.PAGE_COUNT_ASC },
	{ label: "Price: Descending", value: BookSearchOrderBy.PRICE_DESC },
	{ label: "Price: Ascending", value: BookSearchOrderBy.PRICE_ASC },
]